src/devices/cpu/tlcs870/tlcs870.h

```diff
@@ -124,53 +108,234 @@ class tlcs870_device : public cpu_device
-
-	uint8_t m_bitpos;
-	uint8_t m_flagsaffected;
 	uint8_t m_cycles;
-
+	uint16_t m_tmppc;
 	uint32_t  m_addr;
 
 	uint16_t m_F;
 
 	/* CPU registers */
 	uint8_t m_RBS; // register base (4-bits)
 
-	inline uint8_t  RM8 (uint32_t a);
-	inline uint16_t RM16(uint32_t a);
-	inline void WM8 (uint32_t a, uint8_t  v);
-	inline void WM16(uint32_t a, uint16_t v);
-	inline uint8_t  RX8 (uint32_t a, uint32_t base);
-	inline uint16_t RX16(uint32_t a, uint32_t base);
-	inline void WX8 (uint32_t a, uint8_t  v, uint32_t base);
-	inline void WX16(uint32_t a, uint16_t v, uint32_t base);
-	inline uint8_t  READ8();
-	inline uint16_t READ16();
-	void decode();
-	void decode_register_prefix(uint8_t b0);
-	void decode_source(int type, uint16_t val);
-	void decode_dest(uint8_t b0);
+	inline uint8_t  RM8(uint32_t a) { return m_program->read_byte(a); }
+	inline uint16_t RM16(uint32_t a) { return RM8(a) | (RM8((a + 1) & 0xffff) << 8); }
+
+	inline void WM8(uint32_t a, uint8_t  v) { m_program->write_byte(a, v); }
+	inline void WM16(uint32_t a, uint16_t v) { WM8(a, v);    WM8((a + 1) & 0xffff, v >> 8); }
 
-	void setbit_param(uint16_t param_type, uint16_t param, uint8_t bit, bool do_flag);
-	uint8_t getbit_param(uint16_t param_type, uint16_t param);
+	inline uint8_t  RX8(uint32_t a, uint32_t base) { return m_program->read_byte(base | a); }
+	inline uint16_t RX16(uint32_t a, uint32_t base) { return RX8(a, base) | (RX8((a + 1) & 0xffff, base) << 8); }
 
-	uint16_t get_addr(uint16_t param_type, uint16_t param_val);
-	uint16_t get_source_val(uint16_t param_type, uint16_t param_val);
-	void set_dest_val(uint16_t param_type, uint16_t param_val, uint16_t dest_val);
+	inline void WX8(uint32_t a, uint8_t  v, uint32_t base) { m_program->write_byte(base | a, v); }
+	inline void WX16(uint32_t a, uint16_t v, uint32_t base) { WX8(a, v, base);   WX8((a + 1) & 0xffff, v >> 8, base); }
+
+	inline uint8_t  READ8() { uint8_t b0 = RM8(m_addr++); m_addr &= 0xffff; return b0; }
+	inline uint16_t READ16() { uint8_t b0 = READ8(); return b0 | (READ8() << 8); }
+
+	uint16_t get_addr(uint16_t opbyte0, uint16_t val);
 
 	uint8_t get_reg8(int reg);
 	void set_reg8(int reg, uint8_t val);
 	uint16_t get_reg16(int reg);
 	void set_reg16(int reg, uint16_t val);
 
+	static const int FLAG_J = 0x80;
+	static const int FLAG_Z = 0x40;
+	static const int FLAG_C = 0x20;
+	static const int FLAG_H = 0x10;
+
+	inline void clear_JF() { m_F &= ~FLAG_J; };
+	inline void clear_ZF() { m_F &= ~FLAG_Z; };
+	inline void clear_CF() { m_F &= ~FLAG_C; };
+	inline void clear_HF() { m_F &= ~FLAG_H; };
+
+	inline void set_JF() { m_F |= FLAG_J; };
+	inline void set_ZF() { m_F |= FLAG_Z; };
+	inline void set_CF() { m_F |= FLAG_C; };
+	inline void set_HF() { m_F |= FLAG_H; };
+
+	inline int is_JF() { return ((m_F & FLAG_J) ? 1 : 0); };
+	inline int is_ZF() { return ((m_F & FLAG_Z) ? 1 : 0); };
+	inline int is_CF() { return ((m_F & FLAG_C) ? 1 : 0); };
+	inline int is_HF() { return ((m_F & FLAG_H) ? 1 : 0); };
```

Comment:
This group could do to be `const`-qualified as they're not supposed to modify state.